### DISCUSSION TITLE
Discussions are underused. Add the number of open discussions next to it.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#57622] [Discussions] [answered]
### Select Topic Area

Product Feedback

### Body

In my opinion, Discussions are a very nice feature, however they're under used, since many people are just used to Issues and don't know Discussions. I would add the number of Discussions next to the button, like all other buttons. And move it to the left or right of Issues. Example:

```
Code      Discussions (45)      Issues (34)      Pull Requests (11)      Actions      Projects (3)      Wiki      Security      Insights
```

```
Code      Issues (34)      Discussions (45)      Pull Requests (11)      Actions      Projects (3)      Wiki      Security      Insights
```

Reminder to vote to give approval

**Comment by Kashumy** [comment]
Hi 🖐️
@aitor-gomila

**Comment by LindseyB** [comment]
Hi, we tried this out for a short period of time and a number of users read this as a todo list of things rather than an indicator of community health. So, we ended up deciding based on the feedback of a number of users that adding the count here wasn't generally beneficial to communities.

**Reply by ghost** [reply]
Thanks for the quick response.

Todo list of things? Do you think of any other way in order to make Discussions more visisble?

**Reply by annaesvensson** [reply, marked as answer]
For the very short period of time this was available I liked it. I saw the number as an indicator of how much work our community had done and hoped this would invite more users to become active. We want more discussions, more comments, more knowledge exchange and more non-code contributions. But apparently other communities had strong feelings against it, because they saw it as yet another number that "should go down". If there isn't a solution that satisfies everyone, maybe it would be good to have a repository setting, maybe something that communities want to decide on individually.

**Reply by ferdnyc** [reply]
I suppose it's understandable, given that the Issues and Pull Requests counts _are_ figures that you'd want to "work down". They represent backlogs of things that (ideally) would transition to a completed state that removes them from the count — whether that's a fixed issue, a merged PR, etc.

Discussions don't have the same resolution-oriented lifecycle, and shouldn't be taken the same way, but I guess if a count is displayed in the same position in the same way, it's inevitable that it _will_ be taken the same way by many people.

Perhaps this is topic-creep for this particular discussion, and if so I apologize, but I'm curious whether other methods of displaying activity counts have been explored? For example, _inside_ the Discussions tab, perhaps displaying per-category counts, or counts of the number of (active? newly-created? unanswered?) discussions in the past N days... something along those lines?

If nothing else, when _viewing_ a category it'd be nice to know how many discussions are in that category. Counts of Open and Closed Issues or PRs **are** shown on their list pages, and the counts are updated each time the list is re-filtered. A feature along those lines seems valuable for Discussions as well, even if the counter in the tab was too open to misinterpretation.

**Reply by ghost** [reply]
Here's an idea: a red circle indicating new discussions have been *created*. Because right now, it makes the dicussions tab hidden.

**Reply by LindseyB** [reply]
> Here's an idea: a red circle indicating new discussions have been created. Because right now, it makes the dicussions tab hidden.

We did A/B test something very similar to this and it did not perform well.

**Reply by ghost** [reply]
I understand that, what do you suggest? Maybe discussions could be adapted to become issues again with changes (upvote button, etc.)

**Comment by mc12345678** [comment]
I don't know the accessibility by others to the following as I submitted [this ticket](https://support.github.com/ticket/personal/0/2231276), but there are many angles by which a repo owner may want to display some sort of data (any data). Yes, it means that options must be coded whether for the owner or the individual user(s) and yes there will be varied interest. 

Problem right now is that there is no indication of any discussion existing under the tab on discussions, there is no indication of new discussions, no indication that a discussion of interest has additional content, etc. 

I tend to agree just showing the number of discussions doesn't even fit in with the idea of having an area to talk about things. After initially accessing the discussions section, it adds no value. If the number of new discussions were adjacent, then sure maybe. New would have to be relative to each user's "last" visit of the tab. 

But then there is the idea of touches to discussions. For example as one that tries to offer solutions to issues, knowing that content has been added to an existing discussion helps target time and effort. Knowing that such exists from outside the list of items shown on the discussion tab helps again to focus. Email account may not be open to see or have seen notification of new posts to subscribed discussions. 

It's almost as if perhaps a popup on hover stats table may help resolve/target all things. It would be nice if that could just always be displayed but space is always of concern. Perhaps some sort of configuration such that if a user selected just one or maybe two features to be displayed then that would become the icon associated and the hover option still offering the desired data. 

Hmm, now that I think of the hover idea though, not entirely sure how well that would work on mobile/touch devices. Even so, a small clickable icon to do a pop-up..

Somewhat unrelated, but the process that is supporting moving issues to discussions doesn't leave a link in the discussion back to the original issue. This causes the content posted to transfer, but posts elsewhere back to the original issue are not referenced to the new discussion and the new discussion doesn't point to the old issue... A lot of the "conversation" gets lost.

**Comment by ferdnyc** [comment]
@aitor-gomila 

> Somewhat unrelated, but the process that is supporting moving issues to discussions doesn't leave a link in the discussion back to the original issue. This causes the content posted to transfer, but posts elsewhere back to the original issue are not referenced to the new discussion and the new discussion doesn't point to the old issue... A lot of the "conversation" gets lost.

Some of your points there didn't seem to match what I'm seeing in practice, or maybe I'm misinterpreting.

What I've seen, in transferring an [issue](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1628) to a [discussion](https://github.com/GSConnect/gnome-shell-extension-gsconnect/discussions/1630) is:
1. The Discussion doesn't _link_ back to the issue, true, but it does indicate that it was "converted from issue #xxxx" in the sidebar metadata.
2. The Issue is closed and locked on transfer, so there couldn't be any new updates on the issue side after the point of migration.
3. The Issue **does** link directly to the new Discussion, in order to redirect all further conversation there.
4. The original Issue posting and all comments up to the point of migration get copied to the newly-created Discussion, so there's really nothing I can see that "gets lost" in the Issue, when it's migrated.

**Reply by ferdnyc** [reply]
Augh! I got tricked _again_ by the "highlighting some text and typing 'r' starts a quote-reply, but _not_ as a reply to the comment that contained the highlighted text" bug.

**Reply by ferdnyc** [reply]
Oh, wait, now I think I understand, sorry. You're talking about links _to_ the original issue, made before it was converted. (Or after, I suppose.)

...I can see how that might be a concern, if the original Issue had existed for a long time and been referenced in many other Issues/PRs before it was migrated. I think the assumption is that, if an Issue is going to be migrated to a Discussion, it'll likely happen fairly soon after they're created — but that's not necessarily a good assumption.

Even if there are other links to the original Issue, though, like I said that issue **does** get linked to the new Discussion, so really it's just an extra level of indirection to follow that link through to the Discussion where all of the original Issue's content can be updated / responded to.

Granted, the "pingback" notes in the Issue history don't get re-connected, but (given that they're created from user comments) I'm not sure how they could — automatically rewriting comments to point somewhere else would probably meet with some pushback. Unless maybe the newly-created Discussion were to take over the number of the original Issue it was created from. (Which feels like a whole other can of worms, potentially.)

**Reply by mc12345678** [reply]
Sorry for not taking the appropriate time to properly format this response. 

As said, this topic is, well, of topic to this original discussion. I'm for it being moved (and possibly linked) from this discussion to a new one. 

I find correct terminology use important. The issue from which the discussion is spawned is not *linked* but instead presented as text. On a mobile I found it at the bottom of the entire conversation. 

In regarding the internal history of the original `issue`, while github may be used for any team collaboration, I tend to use it for code development. My historical reference is that an issue is something wrong, it needs resolution other than to just discuss it. Certainly there are cases where one perceives something as an issue, but it is just a misunderstanding. E.g. a result should be given as an `integer`, but the design of the code and end user of the data is a `string`. There are ways to get the `integer` portion of that string if one wants to otherwise get that. To that end, it seems to me that such an issue would be appropriate to convert to a discussion upon resolution of the misunderstanding. 

But that said, the reason I brought up this topic is that the original issue had a proposed functional solution linked to the issue from an associated pull request. Further that pull request was linked early in the issue with multiple items posted by others after that pull request. Now tat the issue has been converted to a discussion, that pull request does not appear in the discussion's time line. It could be now again referenced either from the discussion (with a lot of words to "insert" where in the time line it originally appeared or restated as a still viable solution which may negatively come across) or referenced from the pull request directly to the conversation. The conversation though will not directly show or reflect that history. Only an inquisitive individual would chase down all the links and maybe piece together what had happened. 

Point is, such migration without transfer of the associated linkages causes disjoint and confusion in the new discussion. Mind you, in the case where this has been happening as a sort of startup of trying to use discussions, the statement has been made by those in control that issues will remain issues only if they are something they are considering to actually address (whether a functional solution has been provided or not).

Another problem of not carrying forward existing resolution (pull request) is that other individuals waste their time solving the presumably unresolved issue by redevelopment instead of further comment.

So yes, there is an aspect of repo management dependent on the owner; however, the current implementation of issue movement to discussion aides in that mismanagement. As to the assumption of prompt movement, in this startup of movement and for whatever reason, some of these issues with proposed solution are 2 years old.

**Reply by ferdnyc** [reply]
@mc12345678 Sorry for the delay in responding to this.

...Well, opinions may indeed vary, but IMHO anything that ends up getting referenced from a PR should either _stay_ an issue, or a new one should be _opened_ for that aspect of the discussion. PRs shouldn't be referencing Discussions, they should be referencing Issues, because that's where anything important/actionable enough to be the subject of a PR belongs. So (again, purely IMHO) it's appropriate to break any referential links between PRs and Discussions. That's just part of properly setting the expectations for how the two project areas differ.

**Comment by Kashumy** [comment]
# ByteBinUtils 💪

ByteBinUtils is an experimental programming language for:

| Features                               | Description                                              |
| -------------------------------------- | -------------------------------------------------------- |
| Drawing things like geometries         | Create graphics and shapes using ByteBinUtils            |
| Making functions                       | Define custom functions for your programs                |
| Printing text                          | Output text and messages to the console                  |
| Integration with HTML                  | Use ByteBinUtils with HTML for interactive experiences   |
| HTML Compilation with @USING <compiler> | Compile BshBin code to HTML with the specified compiler  |

## Repository

GitHub Repository: [Kashumy/ByteBinUtils](https://github.com/Kashumy/ByteBinUtils)

## Clone

To clone the repository, use the following command:

```bash
git clone https://github.com/Kashumy/ByteBinUtils.git
```

## Getting Started

To get started with ByteBinUtils, follow these steps:

1. Install the required dependencies 
on
https://github.com/Kashumy/ByteBinUtils 
3. Use the provided examples to learn and experiment with the language.
4. Refer to the documentation for detailed information on the language's syntax and features on https://github.com/Kashumy/ByteBinUtils

## Contributing

Contributions are welcome! If you would like to contribute to ByteBinUtils, please follow these guidelines:
- Fork the repository and create your branch.
- Make your changes and test them thoroughly.
- Submit a pull request describing your changes and their purpose.

## License

ByteBinUtils is released under the own free open source License like Mit on

https://github.com/Kashumy/ByteBinUtils

Happy coding with ByteBinUtils! 🚀✨
```

**Reply by ferdnyc** [reply]
Spam reported, happy landings.
